### PR TITLE
User/purifoypaul/rs5 packaged com activation

### PR DESCRIPTION
**Pull request by pmpurifoy**
- Plumb packaged com clsid to LRP for CoCreateInstance
- Fix CreateChannelRequest to work with new UDK fixes
- Added logic to parse the correct IBackgroundTaskInstance in packaged/unpackaged scenarios

### CODE REVIEW

**Comment by pmpurifoy** [comment]
This PR needs to be rebased but I wanted to post a current functioning version of RS5 for packaged. Affected files are 
- PushNotificationManager.cpp
- GetNotificationEventArgs.h
- externs.h
- +PushBackgroundTaskInstance.cpp/h
- platform.cpp (clsid plumbing)
- NotificationListener/Manager.cpp

File: dev/PushNotifications/PushNotificationsLongRunningTask/NotificationListener.cpp [hunk 1]
```diff
@@ -47,7 +47,7 @@ STDMETHODIMP_(HRESULT __stdcall) NotificationListener::OnRawNotificationReceived
 
             auto localBackgroundTask = winrt::create_instance<winrt::Windows::ApplicationModel::Background::IBackgroundTask>(m_comServerClsid, CLSCTX_ALL);
             auto pushBackgroundTaskInstance{ winrt::make<PushBackgroundTaskInstance>() };
```
Comment (jonwis) [inline dev/PushNotifications/PushNotificationsLongRunningTask/NotificationListener.cpp:49]: Check out C++/WinRT's `make_self` that returns you the `impl` type you can use, like this:

```c++
auto pushBackgroundTaskInstance { winrt::make_self<PushBackgroundTaskInstance>() };
pushBackgroundTaskInstance->SetRawNotificationPayload(payloadString);
localBackgroundTask.Run(*pushBackgroundTaskInstance);
```

**Review by jonwis — APPROVED** [review]

File: dev/PushNotifications/PushNotifications.idl [hunk 2]
```diff
@@ -39,11 +39,9 @@ namespace Microsoft.Windows.PushNotifications
         // Initialize using a Registration option and optionally defined parameters like manifest defined activatorId
         // 1) If kind = PushTrigger is specified, only the Push Trigger will be Registered with Background Infra
         // 2) If kind = ComActivator is specified, the Background Task component will be Registered as an in-process COM server
+        // 3) If kind = ProtocolActivator is specified, the LRP will CoCreateInstance the application in background scenarios
```
Comment (pmpurifoy) [inline dev/PushNotifications/PushNotifications.idl:42]: Need to revert this change but we should tell packaged devs to call this constructor with protocol activator & comClsid if push trigger is unsupported. Unpackaged apps will be using the other one I removed and we can assume GUID_NULL. What do you think here? @hulumane
